napari/_qt/qt_viewer.py

```diff
@@ -481,17 +479,26 @@ def console(self, console):
 
     @ensure_main_thread
     def _on_slice_ready(self, event):
+        """Callback connected to `viewer._layer_slicer.events.ready`.
+
+        Provides updates after slicing using the slice response data.
+        This only gets triggered on the async slicing path.
+        """
         responses = event.value
         logging.debug('QtViewer._on_slice_ready: %s', responses)
         for layer, response in responses.items():
             # Update the layer slice state to temporarily support behavior
             # that depends on it.
             layer._update_slice_response(response)
-            # The rest of `Layer.refresh` after `set_view_slice`, where `set_data`
-            # notifies the corresponding vispy layer of the new slice.
+            # The rest of `Layer.refresh` after `set_view_slice`, where
+            # `set_data` notifies the corresponding vispy layer of the new
+            # slice.
             layer.events.set_data()
             layer._update_thumbnail()
             layer._set_highlight(force=True)
+            # Update the layer's loaded state after everything else, so
```

Comment:
@jni : would be useful to get your feedback on the ordering here (i.e. after update the `loaded` state after everything else slice related), since this call may also trigger `Layer.events.loaded` to be emitted

Reply:
I don't understand the full consequences of your question... If the VisPy layer no longer needs that value (?), then yes, doing it at the end seems like the most intuitive thing...